code/modules/paperwork/origami.dm

```diff
@@ -0,0 +1,148 @@
+	internal_paper_tmp.forceMove(loc)
+	internalPaper = null
+	qdel(src)
+	user.put_in_hands(internal_paper_tmp)
+
+/obj/item/origami/attackby(obj/item/P, mob/living/carbon/human/user, params)
+	..()
+	if(istype(P, /obj/item/pen) || istype(P, /obj/item/toy/crayon))
+		to_chat(user, "<span class='notice'>You should unfold [src] before changing it.</span>")
+		return
+
+	else if(istype(P, /obj/item/stamp)) 	//we don't randomize stamps on origami
+		internalPaper.attackby(P, user) //spoofed attack to update internal paper.
+		update_icon()
+
+	else if(P.is_hot())
+		if(HAS_TRAIT(user, TRAIT_CLUMSY) && prob(10))
+			user.visible_message("<span class='warning'>[user] accidentally ignites [user.p_them()]self!</span>", \
+				"<span class='userdanger'>You miss [src] and accidentally light yourself on fire!</span>")
+			user.dropItemToGround(P)
+			user.adjust_fire_stacks(1)
+			user.IgniteMob()
+			return
+
+		if(!(in_range(user, src))) //to prevent issues as a result of telepathically lighting a paper
+			return
+		user.dropItemToGround(src)
+		user.visible_message("<span class='danger'>[user] lights [src] ablaze with [P]!</span>", "<span class='danger'>You light [src] on fire!</span>")
+		fire_act()
+
+	add_fingerprint(user)
+
+/obj/item/origami/papercrane
+	name = "paper crane"
+	desc = "Paper folded to look like a majestic crane."
+	icon_state = "papercrane"
+
+/obj/item/origami/paperboat
+	name = "paper boat"
+	desc = "Paper folded to look like a small boat."
+	icon_state = "paperboat"
+
+/obj/item/origami/paperfrog
+	name = "paper frog"
+	desc = "Paper folded to look like a frog."
+	icon_state = "paperfrog"
+
+/obj/item/origami/papersyndicate
+	name = "paper S"
+	desc = "Paper folded into the letter S. Could this be the work of a syndicate agent?"
+	icon_state = "papersyndicate"
+
+/obj/item/paper/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Alt-click [src] to fold it into origami.</span>"
+
+/obj/item/paper/AltClick(mob/living/carbon/user, obj/item/I)
+	if(!istype(user) || !user.canUseTopic(src, BE_CLOSE, ismonkey(user)))
+		return
+
+	var/list/radial_list = list(
+		"Paper plane" = image(icon = 'icons/obj/bureaucracy.dmi', icon_state = "paperplane"),
+		"Paper crane" = image(icon = 'icons/obj/bureaucracy.dmi', icon_state = "papercrane"),
+		"Paper frog" = image(icon = 'icons/obj/bureaucracy.dmi', icon_state = "paperfrog"),
+		"Paper boat" = image(icon = 'icons/obj/bureaucracy.dmi', icon_state = "paperboat")
+	)
+
+	var/datum/action/innate/origami/origami_action = locate() in user.actions
+	if(origami_action)
+		//Origami Master
+		radial_list["Syndicate paper plane"] = image(icon = 'icons/obj/bureaucracy.dmi', icon_state = "paperplanesyndicate")
+		radial_list["Paper S"] = image(icon = 'icons/obj/bureaucracy.dmi', icon_state = "papersyndicate")
+
+	var/origami_selected = show_radial_menu(user, src, radial_list, require_near = TRUE, tooltips = TRUE)
+	if(!origami_selected || !user || user.stat == DEAD)
```

Comment:
Should probably check if(user.stat) since they probably shouldn't be able to fold origami while unconcious

Reply:
So replace user.stat == DEAD with just !user.stat? I don't really know what user.stat is

Reply:
User.stat is their living status, user.stat means they are in credit, dead or unconscious and !user.stat means they are alive because of the way its coded. You could also do user.stat != ALIVE or whatever it is

Reply:
I fucked up